lua-tree/share/lua/5.1/kong/db/schema/init.lua

```diff
@@ -55,6 +54,23 @@ do
   end
 end
 
+-- `table.nkeys()` is a LuaJIT function and is not available to the Lua VM that goks uses.
+local nkeys
+do
+  local ok
+  ok, nkeys = pcall(require, "table.nkeys")
+  if not ok then
+    nkeys = function (tab)
+      local count = 0
+      for _, v in pairs(tab) do
+        if v ~= nil then
+          count = count + 1
+        end
+      end
+      return count
+    end
+  end
+end
```

Comment:
This is copypasta from [resty/healthecheck.lua](https://github.com/Kong/goks/blob/main/lua-tree/share/lua/5.1/resty/healthcheck.lua#L71-L83). I'm following the pattern of `local_tab` initialized on L46 of this file. This addresses the error: 
```
lua-tree/share/lua/5.1/kong/db/schema/init.lua:4: module table.nkeys not found:
    no field package.preload['table.nkeys']
    open /table/nkeys.lua: file does not exist
    open lua-tree/share/lua/5.1/table/nkeys.lua: file does not exist
    open lua-tree/share/lua/5.1/table/nkeys/init.lua: file does not exist,
```
I can't tell if this resolves the issue, however. I get a new error:
```
Received unexpected error:
lua-tree/share/lua/5.1/kong/constants.lua:222: table index is nil
stack traceback:
    lua-tree/share/lua/5.1/kong/constants.lua:222: in function <lua-tree/share/lua/5.1/kong/constants.lua:0>
    [G]: in function 'require'
    lua-tree/share/lua/5.1/kong/db/schema/typedefs.lua:6: in function <lua-tree/share/lua/5.1/kong/db/schema/typedefs.lua:0>
    [G]: in function 'require'
    setup.lua:6: in main chunk
    [G]: ?
```
I'm able to validate the method is called. The if block beginning on line 62 is entered but `nkeys = function (tab)` on line 63 panics. I'm not sure if it's the method call itself or something else. A print statement on line 64 is never executed. Curiously, print statements[ before the method is called](https://github.com/Kong/kong-ee/blob/master/kong/db/schema/init.lua#L1814) are also not executed?

The stacktrace is also not very illuminating: `lua-tree/share/lua/5.1/kong/constants.lua:222: in function <lua-tree/share/lua/5.1/kong/constants.lua:0>` is on a different line in [kong-ee ](https://github.com/Kong/kong-ee/blob/master/kong/constants.lua#L264) and points to `LOG_LEVELS`. This looks new to 3.1, I don't see it in this [3.0 branch](https://github.com/Kong/kong-ee/blob/next/3.0.x.x/kong/constants.lua).

My guess is either:
1. This is a new error.
2. The update I made to `nkeys` somehow breaks this import, or something that processes this import.

Number 1 is the most reasonable, but I don't really understand what `table index is nil` means in this context and I'm struggling to trace where exactly this is breaking.

Reply:
> I don't really understand what `table index is nil` means in this context and I'm struggling to trace where exactly this is breaking.

the `LOG_LEVELS` table, starting in `constants.lua:222` has this form:
```lua
  LOG_LEVELS = {
    debug = ngx.DEBUG,
    info = ngx.INFO,
    notice = ngx.NOTICE,
    warn = ngx.WARN,
    error = ngx.ERR,
    crit = ngx.CRIT,
    alert = ngx.ALERT,
    emerg = ngx.EMERG,
    [ngx.DEBUG] = "debug",
    [ngx.INFO] = "info",
    [ngx.NOTICE] = "notice",
    [ngx.WARN] = "warn",
    [ngx.ERR] = "error",
    [ngx.CRIT] = "crit",
    [ngx.ALERT] = "alert",
    [ngx.EMERG] = "emerg",
  },
```
lines 231-238 use the form `[key-expression] = value-expression`, so it evaluates the `ngx.DEBUG`, `ngx.INFO`, etc to get the actual keys  (unlike `key-name=value-expression`, which directly sets the key as a string constant).  if any of these is `nil`, this error will trigger.  unfortunately, this aborts the whole (sub) table construction, so we don't know which of these is failing.

Now to find where are those `ngx.XXXX` values....

Reply:
> Now to find where are those `ngx.XXXX` values....

Can't find them.  I think the easiest would be to add to `internal/vm/setup.go`

Reply:
[ff8aeb1](https://github.com/Kong/goks/pull/36/commits/ff8aeb1caa97562ac7e15424a00f176a48ad5f03)